verilog/formatting/formatter.cc

```diff
@@ -230,15 +230,265 @@ static void DeterminePartitionExpansion(partition_node_type* node,
       break;
     }
     case PartitionPolicyEnum::kFitOnLineElseExpand: {
-      if (verible::FitsOnLine(uwline, style)) {
+      if (verible::FitsOnLine(uwline, style).fits) {
         VLOG(3) << "Fits, un-expanding.";
         node_view.Unexpand();
       } else {
         VLOG(3) << "Does not fit, expanding.";
         node_view.Expand();
       }
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+}
+
+//
+// This function formats expanded_lines in a simple greedy manner
+// that is by appending consecutive lines and wrapping only when necessary to stay
+// under the column limit.
+//
+// It operates on lines collection and does not know anything about
+// partitions/subpartition/token partitions and so on.
+//
+// Function takes as argument expanded_lines references, format styling structure and
+// boolean which can be used to force wrapping decision.
+// It returns fitting results by fitted_lines reference and wrapping decision by
+// return value.
+//
+static bool AppendFittingSubpartitions(
+    std::vector<UnwrappedLine>& fitted_lines,
+    const std::vector<UnwrappedLine>& expanded_lines,
+    const FormatStyle& style,
+    bool wrap_second_line) {
+  bool wrapped_second_line = wrap_second_line;
+
+  if (expanded_lines.size() == 0) {
+    fitted_lines.clear();
+    return wrapped_second_line;
+  } else if (expanded_lines.size() == 1) {
+    fitted_lines.push_back(expanded_lines[0]);
+    return wrapped_second_line;
+  } /* else .size() >= 2 */
+
+  // Start current_line at the begging
+  UnwrappedLine current_line(expanded_lines[0].IndentationSpaces(),
+                             expanded_lines[0].TokensRange().begin());
+  int indent;
+
+  // Wraps or not second line depending on wrap_second_line argument and fitting result
+  current_line.SpanUpToToken(expanded_lines[1].TokensRange().end());
+  if (wrap_second_line || (verible::FitsOnLine(current_line, style).fits == false)) {
+    // Forced to wrap
+
+    // Push first line
+    fitted_lines.push_back(expanded_lines[0]);
+
+    // Calculate indenting. Increase current indentation by style-defined wrapping spacing
+    indent = style.wrap_spaces + expanded_lines[0].IndentationSpaces();
+
+    // Start current line right after first line
+    current_line = UnwrappedLine(indent, expanded_lines[0].TokensRange().end());
+
+    // Ensure that current line spans up to end of second line
+    current_line.SpanUpToToken(expanded_lines[1].TokensRange().end());
+
+    // Signalize that first line was wrapped
+    wrapped_second_line = true;
+  } else {
+    // Don't have to wrap second line
+
+    // Compute new indentation level based on first line length
+    indent = verible::FitsOnLine(expanded_lines[0], style).final_column;
+
+    // current_line already spans up to end of second line
+  }
+
+  for (size_t idx = 2 ; idx < expanded_lines.size() ; ++idx) {
```

Comment:
// The remaining lines are appended as long as they fit, wrapping to a position determined by `expanded_lines[1]`.

Reply:
Done